core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -179,7 +179,13 @@ void *impl_allocate_common(const int device_id,
 #ifndef CUDART_VERSION
 #error CUDART_VERSION undefined!
 #elif (defined(KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC) && CUDART_VERSION >= 11020)
-  if (arg_alloc_size >= memory_threshold_g) {
+  int device_supports_cuda_malloc_async;
```

Comment:
Do we have a feel for what GPUs do not support it?
Should we be planning for dropping that check when/if it becomes unnecessary?

Reply:
According to https://github.com/kokkos/kokkos-core-wiki/pull/525#discussion_r1603636387, it's difficult to know.

Reply:
Maybe say we can probably get rid of it next time we bump our requirements and drop support for Kepler

Reply:
Sure